tools/testbench/utils.c

```diff
@@ -388,10 +389,121 @@ static int tb_parse_amixer(struct testbench_prm *tp, char *line)
 	return ret;
 }
 
+static int tb_parse_sofctl(struct testbench_prm *tp, char *line)
+{
+	char control_name[TB_MAX_CTL_NAME_CHARS] = {0};
+	char blob_name[TB_MAX_BLOB_NAME_CHARS] = {0};
+	struct tb_ctl *ctl;
+	uint32_t *blob_bin;
+	char *blob_str;
+	char *end;
+	char *find_ctl_name_str = "-c name=\"";
+	char *find_end_str = "\" ";
+	char *find_set_switch_str = "-s";
+	char *name_str;
+	char *rest;
+	char *token;
+	int copy_len;
+	int find_len = strlen(find_ctl_name_str);
+	int n = 0;
+	int ret = 0;
+	FILE *fh;
+
+	name_str = strstr(line, find_ctl_name_str);
+	if (!name_str) {
+		fprintf(stderr, "error: no control name in script line: %s\n", line);
+		return -EINVAL;
+	}
+
+	end = strstr(&name_str[find_len], find_end_str);
+	if (!end) {
+		fprintf(stderr, "error: no control name end quote in script line: %s\n", line);
+		return -EINVAL;
+	}
+
+	copy_len = end - name_str - find_len;
+	memcpy(control_name, name_str + find_len, copy_len);
```

Comment:
is it guaranteed, that `control_name` is long enough?

Reply:
I don't know if there's a limit topology, likely there is. I guess I should use strdup() and allocate these dynamically.